wasmdome/src/main.rs

```diff
@@ -29,50 +32,215 @@ struct CliCommand {
     actor_paths: Vec<PathBuf>,
 }
 
-fn preconfigure_host() -> std::result::Result<(), Box<dyn std::error::Error>> {
+fn preconfigure_host() -> std::result::Result<WasccHost, Box<dyn std::error::Error>> {
     let host = WasccHost::new();
+    let nats = NatsProvider::new();
+    let kv = KeyvalueProvider::new();
+    let streams = InmemoryStreamsProvider::new();
+    let logger = LoggingProvider::new();
+    host.add_native_capability(NativeCapability::from_instance(nats, None)?)?;
+    host.add_native_capability(NativeCapability::from_instance(kv, None)?)?;
+    host.add_native_capability(NativeCapability::from_instance(streams, None)?)?;
+    host.add_native_capability(NativeCapability::from_instance(logger, None)?)?;
+
     // Load
-    // 1 - real NATS
-    // 2 - in-memory K/V
-    // 3 - in-memory streams
-    host.add_native_capability(NativeCapability::from_file("./libnats_provider.so", None)?)?;
-    host.add_native_capability(NativeCapability::from_file("./libkeyvalue.so", None)?)?;
-    host.add_native_capability(NativeCapability::from_file(
-        "./libtest_streams_provider.so",
+    // 1 - Command Processor (messaging + inmemory keyvalue + logging)
+    // 2 - Match Coordinator (messaging + inmemory keyvalue + logging + extras)
+    // 3 - Historian (messaging + streams + logging)
+    host.add_actor(Actor::from_file("./command_processor_signed.wasm")?)?;
+    host.bind_actor(
+        "MA45FJLFLFWCAYQERFKYYOSRT522ZBOZ3RHFS5BNP5SDI4WIM3QAYBIN",
```

Comment:
Take a look at the pattern in the most recent PR in the [gantry](https://github.com/wascc/gantry) repo. Instead of binding by hard-coded string, we should be able to query the `public_key()` of each actor we load. This may mean not using the manifest, but it will at least let people run the CLI that don't have the same signing keys we do.

Reply:
One option we might want to consider is embedding the raw bytes of the **release** version of each of the core actors (the ones required to resolve a match) via the `bytes!` macro, this way all of those actors are pre-compiled into the wasmdome CLI and you can just load them in via `Actor::from_bytes` ... this would truly allow the CLI to be a standalone binary, hopefully one people could install via `cargo install wasmdome`

Reply:
As an example, a makefile on the CLI could pre-invoke the release make option for all of the other actors and that way have the latest binary embedded for each of the actors.

Reply:
Does this differ from just invoking the `make release` option, and then loading all of the modules with Actor::from_file? As opposed to including the bytes and then loading from bytes.

Reply:
From file requires the file to be present on the developers machine.
Include bytes embeds the bytes in tbe source code and therefore in tbe CLI
binary